engine/src/lib.rs

```diff
@@ -21,3 +21,300 @@ pub mod witness;
 pub mod btc;
 pub mod dot;
 pub mod eth;
+
+use crate::state_chain_observer::client::CreateStateChainClientError;
+use ::multisig::{bitcoin::BtcSigning, eth::EthSigning, polkadot::PolkadotSigning};
+use cf_primitives::CfeCompatibility;
+use engine_proc_macros::cfe_entrypoint;
+use state_chain_observer::client::{
+	chain_api::ChainApi, extrinsic_api::signed::SignedExtrinsicApi, storage_api::StorageApi,
+	STATE_CHAIN_CONNECTION,
+};
+
+use self::{
+	btc::retry_rpc::BtcRetryRpcClient,
+	db::{KeyStore, PersistentKeyDB},
+	dot::retry_rpc::DotRetryRpcClient,
+	eth::{retry_rpc::EthRetryRpcClient, rpc::EthRpcSigningClient},
+	settings::{CommandLineOptions, Settings, DEFAULT_SETTINGS_DIR},
+};
+use anyhow::Context;
+use cf_chains::dot::PolkadotHash;
+use cf_primitives::AccountRole;
+use chainflip_node::chain_spec::use_chainflip_account_id_encoding;
+use clap::Parser;
+use engine_upgrade_utils::{
+	rust_string_args, ExitStatus, ERROR_READING_SETTINGS, NO_START_FROM, SUCCESS,
+};
+
+use futures::FutureExt;
+use libc::c_char;
+use std::{
+	sync::{atomic::AtomicBool, Arc},
+	time::Duration,
+};
+use utilities::{cached_stream::CachedStream, metrics, task_scope::task_scope};
+
+// The `cfe_entrypoint` macro adds the required C parameters to the function signature
+#[cfe_entrypoint]
```

Comment:
Most of this file is just moved to lib, as main is no longer required. The main change is using this macro to generate the entrypoint.